lib/salus/scanners/report_go_dep.rb

```diff
@@ -7,38 +7,126 @@
-      unless should_run?
+      if @repository.go_sum_present?
+        record_dep_from_go_sum
+      elsif @repository.go_mod_present?
+        record_dep_from_go_mod
+      elsif @repository.dep_lock_present?
+        record_dep_from_go_lock_package
+      else
         raise(
           InvalidScannerInvocationError,
-          'Cannot report on Go dependencies without a Gopkg.lock file.'
+          'Cannot report on Go dependencies without a Gopkg.lock, go.mod, or go.sum file'
         )
       end
+    end
+
+    def record_dep_from_go_sum
+      go_sum_path = "#{@repository.path_to_repo}/go.sum"
+      dep_list = []
+
+      File.open(go_sum_path).each("=\n") do |line|
+        dep_list.append(
+          {
+            "fullDependency" => line,
+          "name" => get_name(line),
+          "version" => get_version(line)
+          }
+        )
+      end
+
+      dep_list.each do |dependency|
+        record_dep_package(
+          name: dependency['name'],
+          reference: "N/A for go.mod/go.sum dependencies",
+          version_tag: dependency['version'],
+          dependency_file: "go.sum",
+          type: "go_sum"
+        )
+      end
+    end
+
+    def record_dep_from_go_mod
+      go_mod_path = "#{@repository.path_to_repo}/go.mod"
+      dep_list = []
+      parse_line = false
+
+      File.open(go_mod_path).each("\n") do |line|
+        if line.include? "require ("
+          parse_line = true
+        # Edge case with only 1 dependency in go.mod
+        elsif line.include? "require "
+          split_dep_list = line.split(' ')
+          dep_list.append(
+            {
+              "fullDependency" => split_dep_list[1] + split_dep_list[2],
+              "name" => split_dep_list[1],
+              "version" => split_dep_list[2]
+            }
+          )
+        elsif line.include? ")"
+          parse_line = false
+        elsif parse_line
+          dep_list.append(
+            {
+              "fullDependency" => line,
+              "name" => get_name(line),
+              "version" => get_version(line)
+            }
+          )
+        end
+      end
+
+      dep_list.each do |dependency|
+        record_dep_package(
+          name: dependency['name'],
+          reference: "N/A for go.mod/go.sum dependencies",
+          version_tag: dependency['version'],
+          dependency_file: "go.mod",
+          type: "go_mod"
+        )
+      end
+    end
+
+    def record_dep_from_go_lock_package
       dep_lock = TOML::Parser.new(@repository.dep_lock).parsed
 
       # Report dependencies
       dep_lock['projects'].each do |dependency|
         record_dep_package(
           name: dependency['name'],
           reference: dependency['revision'],
-          version_tag: dependency['version']
+          version_tag: dependency['version'],
+          dependency_file: "Gopkg.lock",
+          type: "go_dep_lock"
         )
       end
     end
 
-    def should_run?
-      @repository.dep_lock_present?
```

Comment:
Any reason not to expand this to include the other checks vs dropping it?

Reply:
https://github.com/coinbase/salus/blob/23cef116093ab6736aaea48be8e55ddea17f6c07/lib/salus/processor.rb#L77. This is called outside of the logic here as part of our interface the scanner should implement

Reply:
Actually I think I deleted this and rewrote it with the other checks around line 121 in report_go_dep.rb :sweat_smile:

Reply:
ah thanks!